src/dlgConnectionProfiles.cpp

```diff
@@ -622,6 +550,109 @@ void dlgConnectionProfiles::slot_deleteProfile()
     delete_profile_dialog->raise();
 }
 
+void dlgConnectionProfiles::validateProfile()
+{
+    bool valid = true;
+
+    notificationArea->hide();
+    notificationAreaIconLabelWarning->hide();
+    notificationAreaIconLabelError->hide();
+    notificationAreaIconLabelInformation->hide();
+    notificationAreaMessageBox->setText(tr(""));
+
+    QListWidgetItem* pItem = profiles_tree_widget->currentItem();
+
+    if (pItem) {
+        QString port = port_entry->text().trimmed();
+        if (port.indexOf(QRegularExpression(QStringLiteral("^\\d+$")), 0) == -1) {
+            QString val = port;
+            val.chop(1);
+            port_entry->setText(val);
+            notificationAreaIconLabelError->show();
+            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(),tr("You have to enter a number. Other characters are not permitted.")));
+            port_entry->setPalette(mErrorPalette);
+            validPort = false;
+            valid = false;
+        }
+
+        bool ok;
+        int num = port.trimmed().toInt(&ok);
+        if (num > 65536 && ok) {
+            notificationAreaIconLabelError->show();
+            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(),tr("Port number must be above zero and below 65535.\r\n")));
+            port_entry->setPalette(mErrorPalette);
+            validPort = false;
+            valid = false;
+        }
+
+#if defined(QT_NO_SSL)
+        if (port_ssl_tsl->isChecked() && port_ssl_tsl->isEnabled())
+        {
+#else
+        if (port_ssl_tsl->isChecked() && !QSslSocket::supportsSsl())
+        {
+#endif
+            notificationAreaIconLabelError->show();
+            notificationAreaMessageBox->setText(QString("%1\n%2").arg(notificationAreaMessageBox->text(),tr("Mudlet is not Configured for SSL connections.\n\n")));
```

Comment:
Could you make this be a tooltip on the checkbox instead? That would be more intuitive (and userfriendly).

Something is wrong with the logic though, for me building Mudlet from source it says SSL is not configured?

Reply:
Yeah, i can, but i put it in notification area, in case a user copies from one computer to another, or the computer looses SSL libraries. It'll be obvious with a tool tip during creation, but not afterwards. Thoughts?

I'm not seeing what's mixed up in logic, but i can split out not compiled with SSL support, and can not find SSL libraries.

Reply:
I think it will still work OK.

Also logic is OK, turns out my Qt libraries really don't have SSL support - so it worked as it needed to.

Reply:
See my comment above - was it a Windows OS? :grinning:

Reply:
No, Ubuntu. Qt is linking against the wrong openssl version.